column.go

```diff
@@ -75,6 +88,39 @@ func renameColumn(name string, newName string, options []ColumnOption) Column {
 	return column
 }
 
+func alterColumnType(name string, typ ColumnType, options []ColumnOption) []Column {
+	column := Column{
+		Op:     SchemaAlter,
+		Name:   name,
+		Type:   typ,
+		Constr: AlterColumnType,
+	}
+	for _, option := range options {
+		if option.isConstraint() {
```

Comment:
i guess you can also do `_, ok := option.(interface { isConstraint() })` here, so no need to implement isConstraint everywhere
but since this is internal api, this also fine

Reply:
I think current way is more explicit and hidden less magic involved :)